kanalony-storage-logic/src/main/scala/kanalony/storage/logic/QueryBase.scala

```diff
@@ -13,23 +13,66 @@ abstract class QueryBase[TReq, TQueryRow] extends IQuery {
 
   val dbApi = DbClientFactory
 
-  protected def extractParams(queryParams : QueryParams) : TReq
+  val metricValueHeaderName = "count"
+
+  private[logic] def extractParams(queryParams : QueryParams) : TReq
 
   private[logic] def executeQuery(params : TReq) : Future[List[TQueryRow]]
 
-  protected def getResultHeaders() : List[String]
+  private[logic] def getResultHeaders() : List[String]
```

Comment:
this method should get the query params as an argument so we can add the metric(s) to the headers list.